enhancements/quota-management/README.md

```diff
@@ -0,0 +1,125 @@
+---
+title: quota-management
+authors:
+  - Lars Kellogg-Stedman <lars@redhat.com>
+reviewers:
+approvers:
+api-approvers:
+creation-date: 2025-09-11
+last-updated: 2025-09-11
+tracking-link:
+see-also:
+replaces:
+superseded-by:
+---
+
+# Quota management
+
+## Summary
+
+We need to implement a quota mechanism for resources allocated by the Open Sovereign AI Cloud solution (oSAC). Our initial implementation will need to integrate with [ColdFront], the resource allocation management tool used by the MOC to manage resource quotas across multiple services.
+
+[coldfront]: https://docs.ccr.buffalo.edu/en/latest/portals/coldfront/
+
+## Motivation
+
+Service providers establish upper limits on the resources tenants can consume for multiple reasons:
+
+- They "sell" chunks of resources (storage, memory, etc) and want to limit tenants to only consume their allocated resources.
+
+- They want to ensure that a tenant cannot accidentally or maliciously consume excessive resources and prevent other tenants from using the service.
+
+### User Stories
+
+- As a service provider, I want to create, modify, and delete resource quotas for a tenant organizations.
+- As a service provider, I want to view quota limits and usage for tenant organizations.
+- As a tenant, I want to be able to view the resource quota limits and utilization that apply to my organization.
+- As a tenant, I want to know when an operation fails due to quota enforcement.
+
+### Goals
+
+TBD.
+
+### Non-Goals
+
+This work is specifically about creating an API and implementation logic for supporting quotas. It does not include any user interface components.
+
+## Proposal
+
+We propose to implement a generic approval workflow in the fulfillment API and to introduce a loosely coupled quota service that will serve as the integration point with ColdFront. When resources are requested via the fulfillment API, they will initially be in an "unapproved" state. The quota service will watch for resources in this state, and for each resource will query the fulfillment service for resource usage information and either approve the resource request or annotate with information about why it was not approved.
+
+The quota service will expose an API that can be used to create, modify, list, and get details for quotas. Write operations (create, modify) will require administrative privileges. Read operations (list, get) will permit a tenant to see quotas associated with their organization and will permit an administrator to see values for all tenants.
+
+Quota limits will be limited as sets of arbitrary key/value pairs, in order to support quotas for new resource types int he future without requiring code changes to the quota service.
+
+There are several advantages to this model:
+
+- The generic approval workflow can be used for things other than quota management. For example, a service provider may require a manual approval workflow for certain types of requests.
```

Comment:
To better support this workflow, I suggest replacing the bool/string API field `approved`, with a list called `approved_by`. This way you can require different kinds of approvers for different kinds of requests, configured in each of the requests (classes?) themselves.

For example a simple cluster may only require `approved_by: ["quota_service"]` whereas others may require `approved_by: ["quota_service", "admin"]`.

PS. Don't want to bikeshed on naming of an API field, just think that a list here better supports the idea of multiple "approval" services that are independent of the fulfillment service from working together.